core/server/themes/loader.js

```diff
@@ -1,40 +1,50 @@
 var debug = require('debug')('ghost:themes:loader'),
     config = require('../config'),
     events = require('../events'),
+    list = require('./list'),
     read = require('./read'),
-    settingsApi = require('../api/settings'),
     settingsCache = require('../settings/cache'),
-    updateConfigAndCache,
-    loadThemes,
+    updateThemeList,
+    loadAllThemes,
+    loadOneTheme,
     initThemes;
 
-updateConfigAndCache = function updateConfigAndCache(themes) {
+updateThemeList = function updateThemeList(themes) {
     debug('loading themes', Object.keys(themes));
-    config.set('paths:availableThemes', themes);
-    settingsApi.updateSettingsCache();
+    list.init(themes);
 };
 
-loadThemes = function loadThemes() {
+loadAllThemes = function loadAllThemes() {
     return read
         .all(config.getContentPath('themes'))
-        .then(updateConfigAndCache);
+        .then(updateThemeList);
+};
+
+loadOneTheme = function loadOneTheme(themeName) {
+    return read
+        .one(config.getContentPath('themes'), themeName)
+        .then(function (readThemes) {
+            // @TODO change read one to not return a keyed object
+            return list.set(themeName, readThemes[themeName]);
+        });
 };
 
 initThemes = function initThemes() {
     debug('init themes', settingsCache.get('activeTheme'));
 
     // Register a listener for server-start to load all themes
     events.on('server:start', function readAllThemesOnServerStart() {
-        loadThemes();
+        loadAllThemes();
     });
 
     // Just read the active theme for now
     return read
         .one(config.getContentPath('themes'), settingsCache.get('activeTheme'))
-        .then(updateConfigAndCache);
+        .then(updateThemeList);
 };
 
 module.exports = {
     init: initThemes,
-    load: loadThemes
+    loadAllThemes: loadAllThemes,
```

Comment:
`themeUtils.loadAllThemes` is exposed right now but not used anywhere.
So we could only expose `themeUtils.reload.one(name)` for now?
Handshakes also with `themeUtils.list.get` a bit more.

I saw that this is only the beginning of theme lib api, so you could ignore my comment.